src/sys/socket/addr.rs

```diff
@@ -1504,6 +1509,7 @@ impl SockaddrStorage {
         as_sockaddr_in, as_sockaddr_in_mut, SockaddrIn,
         AddressFamily::Inet, libc::sockaddr_in, sin}
 
+    #[cfg(not(any(target_os = "haiku")))]
     #[cfg(feature = "net")]
     accessors!{
         as_sockaddr_in6, as_sockaddr_in6_mut, SockaddrIn6,
```

Comment:
this might have been when trying to trace down some other error messages that were cropping up at the time and somehow managed to sneak in. i have removed this exclusion